### PR TITLE
Llama 3

### PR DESCRIPTION
**Pull request by awaelchli**
Adds the Llama3 config for the HF models.

Example prompts:

```
litgpt chat --checkpoint_dir checkpoints/meta-llama/Meta-Llama-3-8B-Instruct
```

```
>> Prompt: Tell me a joke.
>> Reply: Here's one:

Why couldn't the bicycle stand up by itself?

(Wait for it...)

Because it was two-tired!
```

```
>> Prompt: What is an interesting fact about Paris?
>> Reply: Paris, the City of Light! Here's an interesting fact: Paris has a secret underground tunnel system known as the "Catacombes de Paris" that stretches for over 150 miles (240 kilometers). The tunnels were created in the 18th century by transporting the remains of millions of Parisians from overcrowded cemeteries to the tunnels, where they were stacked in a macabre yet fascinating display. Today, parts of the Catacombes are open to the public for guided tours. Who knew that beneath the City of Romance and Revolution, there was a whole world of hidden history and intrigue?
```

```
>> Prompt: What is the square root of i^3, where i is the imaginary unit.
>> Reply: A nice question!

The imaginary unit i is defined as the square root of -1, i.e., i^2 = -1.

Now, let's calculate i^3:

i^3 = i^2 × i = (-1) × i = -i

So, the square root of i^3 is:

√(-i) = ±√(i) × √(-1)
= ±i^(1/2) × i^(1/2)
= ±i
```

A follow-up PR can add the finetuning configs.

### CODE REVIEW

**Review by williamFalcon — APPROVED** [review]

**Review by lantiga — APPROVED** [review]
Looks great! Cannwe start with the llama 2 config for the config_hub for the time being?

File: litgpt/config.py [hunk 1]
```diff
@@ -836,6 +836,54 @@ def norm_class(self) -> Type:
...
+llama_3 = [
+    # https://huggingface.co/meta-llama/Meta-Llama-3-8B/blob/main/config.json
+    dict(
+        name="Llama-3-8B{}",
+        hf_config=dict(org="meta-llama", name="Meta-Llama-3-8B{}"),
+        vocab_size=128256,
+        padding_multiple=64,
+        n_layer=32,
+        n_head=32,
+        # n_embd=8192,
```
Comment (Andrei-Aksionov) [inline litgpt/config.py:853]: Why is it kept commented out?

Comment (awaelchli) [reply litgpt/config.py:853]:
```suggestion
```

File: tutorials/download_model_weights.md [hunk 2]
```diff
@@ -3,29 +3,30 @@
...
+| StableCode by Stability AI                   | 3B                                      | [Stability AI 2023](https://stability.ai/blog/stablecode-llm-generative-ai-coding)                                       |
+| StableLM by Stability AI                     | 3B, 7B                                  | [Stability AI 2023](https://github.com/Stability-AI/StableLM)                                                            |
+| StableLM Zephyr by Stability AI              | 3B                                      | [Stability AI 2023](https://stability.ai/blog/stablecode-llm-generative-ai-coding)                                       |
+| TinyLlama by Zhang et al.                    | 1.1B                                    | [Zhang et al. 2023](https://github.com/jzhang38/TinyLlama)                                                               |
+| Vicuna by LMSYS                              | 7B, 13B, 33B                            | [Li et al. 2023](https://lmsys.org/blog/2023-03-30-vicuna/)                                                              |
```
Comment (Andrei-Aksionov) [inline tutorials/download_model_weights.md:30]: Are you planning to update the output from 
```bash
litgpt download
```

(right below this table)
?

File: litgpt/config.py [hunk 3]
```diff
@@ -836,6 +836,54 @@ def norm_class(self) -> Type:
...
         configs.append(copy)
+        
+
+###############
+# Meta LLaMA 3
+###############
+llama_3 = [
+    # https://huggingface.co/meta-llama/Meta-Llama-3-8B/blob/main/config.json
+    dict(
+        name="Llama-3-8B{}",
```
[the remark this answers is not in the document]
Comment (awaelchli) [reply litgpt/config.py:847]: They are there for string formatting in L883. It gets replaced either with the empty string, or with "-Instruct".

**Pull request merged**